File: common/src/main/java/org/cleanas2/common/exception/DispositionException.java
package org.cleanas2.common.exception;


import org.cleanas2.common.disposition.DispositionType;

public class DispositionException extends Exception {
    private DispositionType disposition;
    private String text;

    public DispositionException(DispositionType disposition, String text, Throwable cause) {
        super(disposition.toString());
        initCause(cause);
        this.disposition = disposition;
        this.text = text;
    }

    public DispositionException(DispositionType disposition, String text) {
        super(disposition.toString());
        this.disposition = disposition;
        this.text = text;
    }

    public static DispositionException error(String statusDescription, String statusMessage) {
        DispositionType typ = DispositionType.error(statusDescription);
        return new DispositionException(typ, statusMessage);
    }

    public static DispositionException error(String statusDescription, String statusMessage, Throwable e) {
        DispositionType typ = DispositionType.error(statusDescription);
        return new DispositionException(typ, statusMessage, e);
    }

    public DispositionType getDisposition() {
        return disposition;
    }

    public void setDisposition(DispositionType disposition) {
        this.disposition = disposition;
    }

    public String getText() {
        return text;
    }

    public void setText(String string) {
        text = string;
    }

}

File: common/src/main/java/org/cleanas2/common/exception/DispositionException.kt
package org.cleanas2.common.exception


import org.cleanas2.common.disposition.DispositionType

class DispositionException : Exception {
    var disposition: DispositionType? = null
    var text: String? = null

    constructor(disposition: DispositionType, text: String, cause: Throwable) : super(disposition.toString()) {
        initCause(cause)
        this.disposition = disposition
        this.text = text
    }

    constructor(disposition: DispositionType, text: String) : super(disposition.toString()) {
        this.disposition = disposition
        this.text = text
    }

    companion object {

        fun error(statusDescription: String, statusMessage: String): DispositionException {
            val typ = DispositionType.error(statusDescription)
            return DispositionException(typ, statusMessage)
        }

        fun error(statusDescription: String, statusMessage: String, e: Throwable): DispositionException {
            val typ = DispositionType.error(statusDescription)
            return DispositionException(typ, statusMessage, e)
        }
    }

}
